tests/python/PEP8NotebookBearTest.py

```diff
@@ -0,0 +1,133 @@
+from tests.LocalBearTestHelper import verify_local_bear
+from bears.python.PEP8NotebookBear import PEP8NotebookBear
+
+
+def prepare_string(file):
+    """
+    Utility function to prepare Notebook strings for further processing.
+
+    replace() necessary to avoid
+    nbformat.reader.NotJSONError: Notebook does not appear to be JSON
+    which is caused by
+    json.decoder.JSONDecodeError: Invalid control character
+    which is caused by the \n characters inside the input cells
+    """
+    return file.replace("\n\"", "\\n\"")
+
+
+good_file = """{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "x = 'good'"
+   ]
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "Python 3",
+   "language": "python",
+   "name": "python3"
+  },
+  "language_info": {
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.5.2"
+  }
+ },
+ "nbformat": 4,
+ "nbformat_minor": 0
+}
+"""
+
+bad_file = """{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "x =    1  # <-- PEP8 Error"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "x =    1  # <-- PEP8 Error but we are in a Markdown cell"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "1\n",
+      "2\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(x)\n",
+    "print(x + x)\n"
+   ]
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "Python 3",
+   "language": "python",
+   "name": "python3"
+  },
+  "language_info": {
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.5.2"
```

Comment:
I removed the content from the metadata field, it is now much shorter.
When you apply the bear to the example file you see that it will skip the markdown cell and the one where there is no error. However, you do not see in in the test if this is what you meant?